src/optimizer/test/partial-eval.spec.ts

```diff
@@ -151,42 +151,26 @@ const mixedExpressions = [
     { original: "(X * -1 * -2) + (X * -1 * -2)", simplified: "X * 4" },
 ];
 
-const failingAssoc3Expressions = [
-    // The following examples should NOT simplify, but they DO.
-    // This is due to edge cases not considered in the safety conditions of the associative rules
-
-    // The following three cases forgot to check the case when c1 in
-    // (c1 - X) is 0. The safety condition works when c1 is not 0.
-
-    // PROBLEM: X could be MIN, so that 0 - X overflows but
-    // -1 - X = MAX does not
-    { original: "(0 - X) + -1", simplified: "-1 - X" },
-
-    // PROBLEM: Same as before
-    { original: "-1 + (0 - X)", simplified: "-1 - X" },
-
-    // PROBLEM: Same as before
-    { original: "(0 - X) - 1", simplified: "-1 - X" },
-
-    // The following case four cases forgot to check that
-    // c2 * c1 < 0 when c1 > 0 in (X * c1) * c2.
-    // The safety condition requires the strict inequality
-    // |c1| < |c1 * c2| for the case c1 > 0 and c1 * c2 < 0,
-    // while currently has |c1| <= |c1 * c2|
-
-    // PROBLEM: X could be -(MIN/2) = 2^255, so that
+const previousFailingAssoc3Expressions = [
```

Comment:
`previousFailingAssoc3Expressions` won't make sense after some time, the should be descriptive outside the context of the development timeline